enhancements/results-forwarding.md

```diff
@@ -0,0 +1,513 @@
+prove painful in multi-cluster environments:
+
+* `ComplianceCheckResults` and `ComplianceRemediations` are specific to the
+  cluster being scanned. Comparing these CRDs across multiple clusters is
+  cumbersome without some sort of filtering mechanism.
+
+* `ComplianceCheckResults` and `ComplianceRemediations` take space in etcd. This
+  space may be better used by other operation-critical components in
+  resource-constrained deployments. These resources also scale with deployment
+  infrastructure size. This can be a significant aspect to consider when
+  running the Compliance Operator, or providing useful data for evidence and
+  auditing.
+  
+* By using CRDs we depend on etcd, which is not a generic database and doesn't
+  provide all the assurances that one would expect when querying persistent
+  data stores. The following are a few short-comings of using etcd as a
+  persistent store for compliance data:
+  
+  - It's non-trivial to do disaster recovery: An object snapshot contains
+    CRD metadata which contains uniquely generated information (e.g., like
+    UUIDs), making it hard to reproduce and replicate. S0me of this data is
+    also irrelevant to compliance checks. Full etcd restore is a tedious and
+    complicated
+    [process](https://platform9.com/kb/kubernetes/restore-etcd-cluster-from-quorum-loss).
+
+  - Values are limited to a size of 1.5Mb, imposing a limit on the data that
+    can be persisted in etcd. This is especially applicable to evidence
+    storage, which will be covered in a separate [enhancement
+    proposal](https://github.com/ComplianceAsCode/compliance-operator/pull/66).
+
+  - The current CRDs may be very verbose, resulting in large objects, which
+    may interfer with known scalability
+    [limitations](https://kubernetes.io/blog/2020/09/02/scaling-kubernetes-networking-with-endpointslices/#scalability-limitations-of-the-endpoints-api)
+    in Kubernetes, causing performance and scalability issues.
+
+While projects such as [StackRox](https://www.stackrox.io/) have successfully
+integrated Compliance Operator using CRDs as the primary interface, we can
+improve the experience of aggregating results by providing a forwarding
+mechanism.
+
+By sending results to a central store, we'll be able to:
+
+* View compliance results and suggested fixes for multiple clusters in one place.
+
+* Address the limitations mentioned above with etcd by choosing an alternative
+  store (e.g., a relational database or a cloud-managed database.)
+
+The Compliance Operator also stores raw results (Asset Reporting Format or ARF)
+in a persistent volume claim by default. While this is useful, not everyone has
+access to persistent storage. This issue is apparent in shared testing clusters
+where automation attempts to delete the Compliance Operator namespace, only to
+have the request fail because of a failed volume allocation.
+
+#### Gathering Evidence
```

Comment:
This is leftover from the evidence proposal?

Reply:
I decided to leave this part here, just to call out that evidence and results were two different things and we were going to try and solve the evidence problem in a separate enhancement.